presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java

```diff
@@ -129,17 +127,16 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
         boolean splitWaitingForAnyNode = false;
 
         // todo identify if sorting will cause bottleneck
-        List<HostAddress> sortedCandidates = sortedNodes(nodeMap);
```

Comment:
- Actually, wonder why don't we put this into `createNodeMapSupplier`? That will save a lot of CPU.
- Can we also build the consistent hash circle in `createNodeMapSupplier`? In that way:
  - Create a new interface `CandidateNodes` in SPI to represent a container for `Collection<HostAddress>`.
  - Change `getPreferredNodes` interface: `List<HostAddress> getPreferredNodes(CandidateNodes sortedCandidates)`
  - You should have at least 2 implementation for `CandidateNodes`: sorted list and consistent hash circle.

Reply:
1. We want the consistent hashing to be more generic to handle more use cases like affinity routing as well
2. I mentioned in my statement about the time complexity, `For improvement, If we can maintain the hash ring with the node management in Presto...` so definitely, we need to combine those two to reduce the time complexity